File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        maximum_profit = 0
        current_low = float(inf)
        for i in range(len(prices)):
            if prices[i] < current_low:
                current_low = prices[i]
            
            maximum_profit = max(maximum_profit, prices[i] - current_low)
        
        return maximum_profit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int global_max {0};
        int l = 0;
        for (int r = 0; r < prices.size(); r++){
            if (prices[r] < prices[l])
                l = r;
            global_max = max(global_max, prices[r] - prices[l]);
        }
        return global_max;
    }
};